src/LM_alg.jl

```diff
@@ -131,12 +131,7 @@ function LM(
     end
 
     # define model and update ρ
-    mk(d) = begin
-      jprod_residual!(nls, xk, d, JdFk)
-      JdFk .+= Fk
-      # JdFk = Jk * d + Fk
-      return dot(JdFk, JdFk) / 2 + σk * dot(d, d) / 2 + ψ(d)
-    end
+    mk(d) = φ(d) + ψ(d)
```

Comment:
I suggest we restore mk without the σ term an do as in R2 below.

Reply:
Does defining an entirely new function take up a lot of space?

Reply:
oh ok. I thought it would be more time-consuming. Plus it feels like aesthetics but if you want it to match that's fine.